### ISSUE TITLE
glibc_2.22.bb error when building spark7162

### ISSUE THREAD

**Issue by joeusercz** [Taapat/vuplus-openpli-oe-core#4]
Building image for spark7162
WARNING: Getting checksum for stslave-fw SRC_URI entry audio.elf: file not found except in DL_DIR                                             | ETA:  00:00:32
WARNING: Getting checksum for stslave-fw SRC_URI entry video.elf: file not found except in DL_DIR
WARNING: Unable to get checksum for ghostscript SRC_URI entry objarch.h: file could not be found###############                               | ETA:  00:00:07
NOTE: glibc can't be built with -Os, -Os -Wno-error will be used instead.#####################################################                | ETA:  00:00:04
NOTE: glibc can't be built with -Os, -Os -Wno-error will be used instead.
NOTE: glibc can't be built with -Os, -Os -Wno-error will be used instead.
NOTE: glibc can't be built with -Os, -Os -Wno-error will be used instead.
WARNING: Unable to get checksum for vuplus-platform-util-vusolose SRC_URI entry spark7162_vuplus-platform-util.sysvinit: file could not be foundETA:  00:00:00
WARNING: Unable to get checksum for vuplus-platform-util-vusolo2 SRC_URI entry spark7162_vuplus-platform-util.sysvinit: file could not be found
WARNING: Unable to get checksum for vuplus-platform-util-vuduo2 SRC_URI entry spark7162_vuplus-platform-util.sysvinit: file could not be found
Parsing recipes: 100% |#######################################################################################################################| Time: 00:00:30
Parsing of 2197 .bb files complete (0 cached, 2197 parsed). 2749 targets, 383 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION        = "1.28.0"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "Debian-8.3"
TARGET_SYS        = "sh4-oe-linux"
MACHINE           = "spark7162"
DISTRO            = "openpli"
DISTRO_VERSION    = "5"
TUNE_FEATURES     = "sh4"
meta-oe
meta-filesystems
meta-multimedia
meta-networking
meta-python       = "HEAD:73854a05565b30a5ca146ac53959c679b27815aa"
meta              = "HEAD:fd75637d97cc3d669229e59c1d21ce7038fc92d7"
meta-fulan
meta-openpli
meta-vuplus-bsp
meta-local        = "master:055ee308a15391e620fd0cc6f23b4cbcb639075b"

NOTE: Preparing RunQueue
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks

WARNING: Failed to fetch URL http://jerakeen.org/files/PythonDaap-0.7.1.tar.gz, attempting MIRRORS if available
WARNING: Failed to fetch URL http://downloads.sourceforge.net/fuse/fuse-2.9.4.tar.gz, attempting MIRRORS if available
ERROR: Error executing a python function in /mnt/sdc1/vuplus-fulan-openpli-oe-core/openembedded-core/meta/recipes-core/glibc/glibc_2.22.bb:

The stack trace of python calls that resulted in this exception/failure was:
File: '/mnt/sdc1/vuplus-fulan-openpli-oe-core/openembedded-core/meta/recipes-core/glibc/glibc_2.22.bb', lineno: 237, function: <module>
     0233:        bb.utils.unlockfile(lf)
     0234:
     0235:
     0236:
 **\* 0237:do_package_ipk(d)
     0238:
File: '/mnt/sdc1/vuplus-fulan-openpli-oe-core/openembedded-core/meta/recipes-core/glibc/glibc_2.22.bb', lineno: 232, function: do_package_ipk
File: '/mnt/sdc1/vuplus-fulan-openpli-oe-core/openembedded-core/meta/recipes-core/glibc/glibc_2.22.bb', lineno: 29, function: cleanupcontrol
     0025:           file://0011-__ieee754_sqrt-f-are-now-inline-functions-and-call-o.patch \
     0026:           file://0012-Make-ld-version-output-matching-grok-gold-s-output.patch \
     0027:           file://0013-sysdeps-gnu-configure.ac-handle-correctly-libc_cv_ro.patch \
     0028:           file://0014-Add-unused-attribute.patch \
 **\* 0029:           file://0015-When-disabling-SSE-also-make-sure-that-fpmath-is-not.patch \
     0030:           file://0016-yes-within-the-path-sets-wrong-config-variables.patch \
     0031:           file://0017-timezone-re-written-tzselect-as-posix-sh.patch \
     0032:           file://0018-eglibc-Cross-building-and-testing-instructions.patch \
     0033:           file://0019-eglibc-Bring-Eglibc-option-group-infrastructure-to-g.patch \
File: '/mnt/sdc1/vuplus-fulan-openpli-oe-core/bitbake/lib/bb/utils.py', lineno: 685, function: prunedir
     0681:            if os.path.islink(os.path.join(root, name)):
     0682:                os.remove(os.path.join(root, name))
     0683:            else:
     0684:                os.rmdir(os.path.join(root, name))
 **\* 0685:    os.rmdir(topdir)
     0686:
     0687:#
     0688:# Could also use return re.compile("(%s)" % "|".join(map(re.escape, suffixes))).sub(lambda mo: "", var)
     0689:# but thats possibly insane and suffixes is probably going to be small
Exception: OSError: [Errno 39] Directory not empty: '/mnt/sdc1/vuplus-fulan-openpli-oe-core/build/tmp/work/sh4-oe-linux/glibc/2.22-r0/packages-split/glibc-dbg/CONTROL'

ERROR: Function failed: do_package_ipk
ERROR: Logfile of failure stored in: /mnt/sdc1/vuplus-fulan-openpli-oe-core/build/tmp/work/sh4-oe-linux/glibc/2.22-r0/temp/log.do_package_write_ipk.8437
ERROR: Task 1206 (/mnt/sdc1/vuplus-fulan-openpli-oe-core/openembedded-core/meta/recipes-core/glibc/glibc_2.22.bb, do_package_write_ipk) failed with exit code '1'
WARNING: QA Issue: ELF binary '/mnt/sdc1/vuplus-fulan-openpli-oe-core/build/tmp/work/sh4-oe-linux/glibc/2.22-r0/packages-split/nscd/usr/sbin/nscd' has relocations in .text [textrel]

Although it says "OSError: [Errno 39] Directory not empty" the directory "/mnt/sdc1/vuplus-fulan-openpli-oe-core/build/tmp/work/sh4-oe-linux/glibc/2.22-r0/packages-split/glibc-dbg/CONTROL"   IS empty after build attempt.

I can post full log if you need...

**Comment by Taapat** [comment]
Please post full log, but on pastebin.

**Comment by joeusercz** [comment]
I restarted the make, and then received another error, so after your other changes, I removed everything and started again with a complete fresh git clone. This time I received another error. Here is the pastebin link.[http://pastebin.com/juan1k63](url)

**Comment by Taapat** [comment]
I build image on Ubuntu-14.04.
I see you build on Debian-8.3.
Perhaps this is a problem. I don't know, perhaps for your system needs something specific.
By the way, what is your system gcc version? I see in openpli repo that for gcc 5 need additional pathes.

**Comment by joeusercz** [comment]
Yes, it is possible, but strange that the problem is with the spark drivers and those exact same drivers build fine on my system when I build enigma2 with duckbox-developers git...
Unfortunately I have not had time to debug further. I was hoping you would have found something simple in the logs.

> $ gcc -v
> Using built-in specs.
> COLLECT_GCC=gcc
> COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.9/lto-wrapper
> Target: x86_64-linux-gnu
> Configured with: ../src/configure -v --with-pkgversion='Debian 4.9.2-10' --with-bugurl=file:///usr/share/doc/gcc-4.9/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.9 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.9 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.9-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --with-arch-32=i586 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
> Thread model: posix
> gcc version 4.9.2 (Debian 4.9.2-10)

**Comment by Taapat** [comment]
duckbox-developers used lot of rpm packages from stlinux.
On openpli everything is build from source codes.
I assume that something is missing some specific stlinux pach.

Have you tried oe-alliance?
Are oe-alliance building without problem?

**Comment by joeusercz** [comment]
Also, duckbox-developers use a 32bit gcc, while oe-alliance builds a native 64bit version.

Is your Ubuntu 32 or 64 bit?

I tried building OpenSPA about a year ago, but did not succeed - but if I recall, the kernel built ok, it was just a problem with enigma2...

I will try a new oe-alliance build tonight.

**Comment by Taapat** [comment]
My Ubuntu is 64 bit.

**Comment by joeusercz** [comment]
Last night I attempted to build openatv. It failed (gstreamer problem) but it did build the kernel and modules ok first.
Here are some diffs of the work and work-shared directories. Most diffs were just in the pahs, but I am not sure of the easiest way to avoid these diffs???
[https://mega.nz/#!KIwjmaqT!WSaoY_t1Hi4NaJHDkOzhO1wdsQJCO_ncj2zQvl1kJa0](url)
I will try to look at it closer later tonight...
If some other diff would be helpful, let me know.

**Comment by joeusercz** [comment]
After changing the source of drivers to duckbox-developers git, I was able to finally build. (Of course the change caused a lot of small problems which had to be hand tweaked (hacked!).)

But I still do not know the exact reason why it would not build with original settings...

I have installed openwebif and oscam, but have not done a lot of testing yet.

I will close since the original problem is still not clear, I will reopen a new issue when (if?) I find the reason.
